### PR TITLE
README.rst: fix badge url

### PR DESCRIPTION
**Pull request by Mariatta**
- Change to https url
- Fix target url (prev url was 404)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,8 +1,8 @@
 The CPython Developer's Guide
 =============================
 
-.. image:: http://readthedocs.org/projects/cpython-devguide/badge/?version=latest
-   :target: http://cpython-devguide.readthedocs.io/en/latest/?badge=latest
+.. image:: https://readthedocs.org/projects/cpython-devguide/badge/?version=latest
+   :target: https://cpython-devguide.readthedocs.io/?badge=latest
    :alt: Documentation Status latest
```
Comment (berkerpeksag) [inline README.rst]: Since we don't have two versions anymore, I guess we can safely remove "latest" here.

File: README.rst [hunk 2]
```diff
@@ -1,8 +1,8 @@
 The CPython Developer's Guide
 =============================
 
-.. image:: http://readthedocs.org/projects/cpython-devguide/badge/?version=latest
-   :target: http://cpython-devguide.readthedocs.io/en/latest/?badge=latest
+.. image:: https://readthedocs.org/projects/cpython-devguide/badge/?version=latest
+   :target: https://cpython-devguide.readthedocs.io/?badge=latest
```
Comment (berkerpeksag) [inline README.rst]: Same for `?badge=latest`.

File: README.rst [hunk 3]
```diff
@@ -1,8 +1,8 @@
 The CPython Developer's Guide
 =============================
 
-.. image:: http://readthedocs.org/projects/cpython-devguide/badge/?version=latest
-   :target: http://cpython-devguide.readthedocs.io/en/latest/?badge=latest
+.. image:: https://readthedocs.org/projects/cpython-devguide/badge/?version=latest
```
Comment (berkerpeksag) [inline README.rst]: Same for `?version=latest`.

Comment (Mariatta) [inline README.rst]: Done :)

**Comment by Mariatta** [comment]
Thanks for the review, @berkerpeksag :)

**Pull request merged**